### ISSUE TITLE
Merge props returned from function with other props in pattern definition

### ISSUE THREAD

**Issue by whastings** [eoneill/eyeglass-restyle#2]
Currently, if I have code like below, the function's returned props will be tacked on after the rest of the props in the definition, preventing the definition from overriding any of the function's props:

```scss
@function func-that-returns-color() {
  @return (color: blue);
}
restyle-define(foo, (
  restyle-function(func-that-returns-color): (),
  color: red
));

.foo {
  @include restyle(foo);
}

// Output:
.foo {
  color: red;
  color: blue;
}
```

It would be nice if the output of the function was merged in instead of tacked on, with later props in the definition overriding props in the function. That way, the output would instead be `.foo { color: red; }`.

**Comment by eoneill** [comment]
Thanks for reporting.

This is a known issue with the way functions are stored in the styles resolution phase. Functions are treated somewhat like `states` and are moved onto a property on the styles array, rather than an entry in the array.

This will require a refactor of the way functions are processed in `Styles#extractStylesFromDefinition`.

**Comment by eoneill** [comment]
Released in `v1.0.2`
